### PR TITLE
Feature 2

### PR DESCRIPTION
**Pull request by GeorgiDinov**
Refactoring of the bubble sort algorithm implementations and light clean up in abstract sorter class.

### CODE REVIEW

File: src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java [hunk 1]
```diff
@@ -42,40 +39,37 @@ private void bubbleSortImplOne(Integer[] array) {
...
 
-        boolean swapNeeded = true;
-        while (i < n - 1 && swapNeeded) {
-            swapNeeded = false;
-            for (int j = 1; j < n - i; j++) {
-                if (array[j - 1] > array[j]) {
-                    swap(array, (j - 1), j);
-                    swapNeeded = true;
+        while (first < last) {
+            boolean isSwapNeeded = false;
```
Comment (MariyqZlateva) [inline src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java]: May be for better readability, change flag name to "isSorted", and set it to "true".

File: src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java [hunk 2]
```diff
@@ -42,40 +39,37 @@ private void bubbleSortImplOne(Integer[] array) {
...
+            for (int current = 0; current < (last - first); current++) {
+                int next = current + 1;
+                if (array[current] > array[next]) {
+                    swap(array, current, next);
+                    isSwapNeeded = true;
```
Comment (MariyqZlateva) [inline src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java]: If swap was needed, the array is not sorted, and set the flag to false.

File: src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java [hunk 3]
```diff
@@ -42,40 +39,37 @@ private void bubbleSortImplOne(Integer[] array) {
...
                 }
             }
-            if (!swapNeeded) {
+            if (!isSwapNeeded) {
```
Comment (MariyqZlateva) [inline src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java]: If the array is sorted, than brake out of the loop.

**Review by MariyqZlateva — CHANGES_REQUESTED** [review]
I have several suggestions for better readability, please look at them.

File: src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java (hunk 2 above)
Comment (GeorgiDinov) [reply src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java]: Resolved.

File: src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java (hunk 3 above)
Comment (GeorgiDinov) [reply src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java]: Resolved.

File: src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java (hunk 1 above)
Comment (GeorgiDinov) [reply src/com/georgidinov/algorithms/sorting/bubblesort/BubbleSort.java]: Change the name of the flag to 'isSorted' and set it to ' true'

**Comment by GeorgiDinov** [comment]
Made the changes that were suggested.

**Review by MariyqZlateva — APPROVED** [review]
Looking good!

**Pull request merged**